### PR TITLE
add $(element).destroy()

### LINKED ISSUE #57: destroy
how to destroy instances?

### PR DESCRIPTION
**Pull request by thijstriemstra**
fixes #57

### CODE REVIEW

File: src/bootstrap-input-spinner.js [hunk 1]
```diff
@@ -96,6 +104,9 @@
             $original[0].setValue = function (newValue) {
                 setValue(newValue)
             }
+            $original[0].destroy = function () {
```
Comment (V-ed) [inline src/bootstrap-input-spinner.js]: Adding a `destroy` function to the original javascript object would be fine by me, but I would make it more verbose that it does not actually destroys the element, but rather its associated spinner. Therefore, here's my suggestion :

```suggestion
            $original[0].destroySpinner = function () {
```

Of course the call to it at your line 27 would change to :
```js
setTimeout(function () {
    element.destroySpinner()
})
```

That's just my two cents!

Comment (thijstriemstra) [reply src/bootstrap-input-spinner.js]: sounds good, feel free to make the change.

Comment (shaack) [reply src/bootstrap-input-spinner.js]: Yes, great, it should return ti the exact state before creation. But I am uncertain about the syntax. Could be `element.destroyInputSpinner()`, `element.inputSpinner("destroy")` or `element.inputSpinner.destroy()`, or something I missed. What is the best practice here?

Comment (thijstriemstra) [reply src/bootstrap-input-spinner.js]: i agree the string method is nicer.

File: src/bootstrap-input-spinner.js [hunk 2]
```diff
@@ -147,6 +158,13 @@
                 }
             }
 
+            function destroy() {
+                observer.disconnect();
+                resetTimer();
+                $input.off("paste input change focusout");
+                $original.next('div').remove();
```
Comment (villu164) [inline src/bootstrap-input-spinner.js:165]:
```suggestion
                $original.next('div').remove();
                $original.show();
```

**Pull request merged**